### PR TITLE
Report message delivery state changes to Packet Broker

### LINKED ISSUE #4119: Report message delivery state changes to Packet Broker
#### Summary

Report message delivery state changes to Packet Broker

#### Why do we need this?

To inform publishing networks whether the message has been processed or, on failure, what the error is.

This is a necessary feature to interact with networks using LoRaWAN Passive Roaming. 

#### What is already there? What do you see now?

The Packet Broker API since 3.6 supports reporting message delivery state changes.

Both GS and NS return appropriate error codes to PBA.

#### What is missing? What do you want to see?

Report uplink and downlink message delivery state changes back to Packet Broker. 

#### How do you propose to implement this?

For uplink messages, the Home Network responds whether the local Network Server processed the message. For downlink messages, the Forwarder responds whether the local Gateway Server sent the message (and, ideally, a TX acknowledgment has been received).

PBA subscribers should therefore check the GS and NS error codes and report the message delivery state accordingly. 

This can go in one go with #3771

#### How do you propose to test this?

CI and manual

#### Can you do this yourself and submit a Pull Request?

Yes

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Closes #4119 

#### Changes

- Extend internal API with GS reported downlink path and which RX window(s) have been used when successfully scheduling downlink
- Report message delivery state changes to Packet Broker

#### Testing

Unit testing

##### Regressions

None expected

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: api/gatewayserver.proto [hunk 1]
```diff
@@ -69,6 +70,12 @@ service GtwGs {
 message ScheduleDownlinkResponse {
   // The amount of time between the message has been scheduled and it will be transmitted by the gateway.
   google.protobuf.Duration delay = 1 [(gogoproto.stdduration) = true, (gogoproto.nullable) = false, (validate.rules).duration.required = true];
+  // Downlink path chosen by the Gateway Server.
+  DownlinkPath downlink_path = 2;
+  // Whether RX1 has been chosen for the downlink message.
+  bool rx1 = 3;
+  // Whether RX2 has been chosen for the downlink message.
+  bool rx2 = 4;
```
Comment (johanstokking) [inline api/gatewayserver.proto]: Technically, the same downlink message can be scheduled on the same downlink path on both RX1 and RX2: the gateway transmitting the same downlink message twice.

Comment (htdvisser) [reply api/gatewayserver.proto]: If you need to explain it here, you may as well explain it in a comment in the proto file.

File: pkg/packetbrokeragent/observability.go [hunk 2]
```diff
@@ -79,29 +79,65 @@ var pbaMetrics = &messageMetrics{
 			"forwarder_cluster_id",
 		},
 	),
+	uplinkStateReported: metrics.NewContextualCounterVec(
```
Comment (htdvisser) [inline pkg/packetbrokeragent/observability.go:82]: I don't know how useful these metrics are. Keep in mind that this is a large number of time series (one for each combination of the label values `forwarder_net_id` ✖️ `forwarder_tenant_id` ✖️ `forwarder_cluster_id` ✖️ `home_network_net_id` ✖️ `home_network_tenant_id` ✖️ `home_network_cluster_id`).

**Review by htdvisser — APPROVED** [review]

File: pkg/packetbrokeragent/observability.go (hunk 2 above)
Comment (johanstokking) [reply pkg/packetbrokeragent/observability.go:82]: [`8a84781` (#4151)](https://github.com/TheThingsNetwork/lorawan-stack/pull/4151/commits/8a84781a85924d5537ab5803796f006d3f1b227b)

**Review by KrishnaIyer — APPROVED** [review]
GS LGTM

**Review by adriansmares — APPROVED** [review]
Nothing to add.

**Pull request merged**